scene/show-labels-on-layer-in-3d/src/main/java/com/esri/samples/show_labels_on_layer_in_3d/ShowLabelsOnLayerIn3dSample.java

```diff
@@ -0,0 +1,138 @@
+
+      // set title, size, and add the JavaFX scene to stage
+      stage.setTitle("Show Labels on Layer in 3D");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(fxScene);
+      stage.show();
+
+      // create a scene from an ArcGISOnline portal item
+      var portalItem = new PortalItem(
+        new Portal("https://www.arcgis.com", false), "850dfee7d30f4d9da0ebca34a533c169");
+      ArcGISScene scene = new ArcGISScene(portalItem);
+
+      // set the scene to the sceneview, and add the sceneview to the JavaFX stack pane
+      sceneView = new SceneView();
+      sceneView.setArcGISScene(scene);
+      // create a progress indicator and display it to indicate loading is in progress
+      var progressIndicator = new ProgressIndicator();
+      progressIndicator.setVisible(true);
+      stackPane.getChildren().addAll(sceneView, progressIndicator);
+
+      scene.addDoneLoadingListener(() -> {
+        if (scene.getLoadStatus() == LoadStatus.LOADED) {
+          // remove the progress indicator when the scene loads
+          progressIndicator.setVisible(false);
+
+          // filter through the scene's operational layers to find the layer named "Gas",
+          // and find the "Gas Main" sublayer to apply the label definition to
+          scene.getOperationalLayers().stream()
+            .filter(layer -> layer.getName().equals("Gas"))
+            .flatMap(gasLayer -> gasLayer.getSubLayerContents().stream())
+            .forEach(subLayer -> {
+              if (subLayer.getName().equals("Gas Main")) {
+                FeatureLayer featureLayer = (FeatureLayer) subLayer;
+                featureLayer.setLabelsEnabled(true);
+                // clear any existing label definitions on the layer
+                featureLayer.getLabelDefinitions().clear();
+
+                // create a text symbol to set to the label definition
+                var textSymbol = new TextSymbol();
+                textSymbol.setColor(ColorUtil.colorToArgb(Color.ORANGE));
+                textSymbol.setHaloColor(ColorUtil.colorToArgb(Color.WHITE));
+                textSymbol.setHaloWidth(2);
+                textSymbol.setSize(16);
+                // create a label definition with an Arcade expression script
+                var labelDefinition = new LabelDefinition();
+                labelDefinition.setExpression(
+                  new ArcadeLabelExpression("Text($feature.INSTALLATIONDATE, `DD MMM YY`)"));
+                labelDefinition.setPlacement(LabelingPlacement.LINE_ABOVE_ALONG);
+                labelDefinition.setUseCodedValues(true);
+                labelDefinition.setTextSymbol(textSymbol);
+
+                // add the label definition to the feature layer
+                featureLayer.getLabelDefinitions().add(labelDefinition);
+              }
+            });
+        } else {
+          System.out.println("Scene failed to load " + scene.getLoadError().getCause());
```

Comment:
I'm just curious here, as how we handle errors is something that came up a lot in review discussions previously and I'm not sure a consistent pattern was ever really reached or agreed upon. What are you thinking going forward now you have the reins longer term - to have things being printed out rather than in an Alert? Or depending on the situation? I still don't really know what I prefer, I guess for the target developer users printing out is sufficient. I just find it hard to escape being a user looking at the app with the progress indicator spinning but not getting a message as to why. Thinking aloud!